aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -0,0 +1,56 @@
+"""
+NodeJS NPM Workflow
+"""
+
+import hashlib
+
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+from aws_lambda_builders.actions import CopySourceAction
+
+from .actions import NodejsNpmPackAction, NodejsNpmInstallAction
+from .utils import OSUtils
+
+
+def unique_dir(manifest_path):
```

Comment:
why do you need this? for hot-reloading?

If you need a unique directory, you can use `tempfile.mkdtemp()` within the scratch directory.

Reply:
I need this so that if you run multiple workflows for various lambda functions and give me the same scratch dir, I don't end up compiling overlapping packages :) you mentioned wanting to support hot reloading later, so I thought that it's better to have a stable name that depends on the manifest, especially as the copysource command will be wired to that. 

so if a workflow runs for the same manifest, it will try to reuse the same directory to prepare the sources. the dependencies won't be installed here, but in the artifact dir directly

Reply:
First requirement of running multiple lambdas on same scratch_dir won't happen. Caller will setup diff scratch dir for each function and each execution. 

I am okay with the stable name, but the directories are cleared & re-created on each run. So it won't really do much help here. 

I would like to skip this for now as we don't have a use-case now and revisit it later. Thoughts?

Reply:
OK - I misunderstood the purpose of `scratch_dir` then. If it's a new directory every time, it makes things very easy. I'll just hard-code the name of the dir. I fixed this in https://github.com/gojko/aws-lambda-builders/commit/d183932559d05f19899a6176d2d65f5b06f2abf5 - let me know if this makes sense now.

Reply:
Cool, that makes sense